src/ControlSystem/Actions/Initialization.hpp

```diff
@@ -71,7 +72,7 @@ struct Initialize {
         db::get<control_system::Tags::ControlSystemInputs<ControlSystem>>(box);
     ::Initialization::mutate_assign<tags_to_be_initialized>(
         make_not_null(&box), option_holder.averager, option_holder.controller,
-        option_holder.tuner);
+        option_holder.tuner, option_holder.name);
```

Comment:
If you need the name in the box (see comment above) then you can still use `ControlSystem::name` here instead of going through the options (since the name is not an option). You can keep the templating on `ControlSystem` (I think it makes it easier to understand code if templating is consistent. `ControlSystem` has a well-defined protocol, so passing it around as a template parameter is good. You can actually insert more `assert_conforms_to` to validate the template parameter.)

Reply:
Right now the name has to be in the OptionHolder because inside the `FunctionsOfTimeInitialize` tag, we don't pass control systems, we pass OptionHolders. Yet we still needed the name so the OptionHolder seemed a nice place to put this because it succinctly holds all info needed to initialize a control system and functions of time.

However, since the OptionHolder is now templated on a control system, I could add an alias in the OptionHolder `using control_system =` instead of the `name` memeber and then inside the `FunctionsOfTimeInitialize` tag do `std::string name = decltype(option_holder)::control_system::name()` using the static `name()` from the ControlSystem. What do you think?

Reply:
Yea that sounds good. When a class has a template parameter there's no reason not to expose it as a type alias.